src/main/java/org/truffleruby/parser/ast/SymbolParseNode.java

```diff
@@ -62,7 +62,11 @@ public SymbolParseNode(SourceIndexLength position, String name, Encoding encodin
             encoding = USASCIIEncoding.INSTANCE;
         }
 
-        this.rope = StringOperations.encodeRope(name, encoding, cr == null ? CodeRange.CR_UNKNOWN : cr);
+        if (cr == CodeRange.CR_7BIT) {
+            this.rope = RopeOperations.encodeAscii(name, encoding);
+        } else {
+            this.rope = StringOperations.encodeRope(name, encoding, cr == null ? CodeRange.CR_UNKNOWN : cr);
```

Comment:
Should we make `StringOperations.encodeRope` delegate to `encodeAscii` is cr is CR_7BIT?

Reply:
Probably. I think at the time I wrote this I wanted them split, but I can't really see why now.

Reply:
Looking more, I think I didn't do it that way because of the boundary. But I also believe I can remove that now.